pkg/controller/compliancescan/compliancescan_controller.go

```diff
@@ -549,6 +510,19 @@ func gatherResults(r *ReconcileComplianceScan, instance *complianceoperatorv1alp
 		if err != nil {
 			isReady = false
 		}
+
+		result, err = getScanResult(r, foundCM)
```

Comment:
Could you please add a comment to either the function definition or here to the call that err can only ever be set if result is ResultError? Alternatively, we could check for err and if it is set, then we could always return ResultError.